controller/src/controller/models/domain/rsptag.py

```diff
@@ -576,3 +578,86 @@ def subset(
 
         # Return the results.
         return type(self)(tags)
+
+    def by_type(self, image_type: RSPImageType) -> Self:
+        """Return all tags of a given type.
+
+        Parameters
+        ----------
+        image_type
+            RSPImageType to select from collection.
+
+        Returns
+        -------
+        RSPImageCollection
+            Collection of tags matching the given type.
+        """
+        return type(self)(self._by_type[image_type])
+
+    def apply_policy(self, policy: RSPImageFilterPolicy) -> Self:
+        """Apply a filter policy and return the remaining tags.
+
+        Note that this relies on the ordering of types, as subset() does.
```

Comment:
This is an invariant maintained by the class and the caller doesn't need to know about it, so it shouldn't be in the docstring.